content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -173,22 +175,34 @@ class CreateAndIssueMarsVoucherInitiator @JsonConstructor constructor(private va
     lateinit var jsonMarshallingService: JsonMarshallingService
 ```
 
-#### Add the flow implementation
+#### Write your business logic
+
+##### Add the flow implementation
 
 Next you must add the flow implementation to the initiating flow by encapsulating it within the `call` method.
 
 1. Add the `@Suspendable` annotation.
 2. Add the `call` method with the return type `SignedTransactionDigest`.
 3. Parse your parameters using the `mapOfParams` value. Since the parameters the flow will use come in JSON format, the JSON object must be parsed to extract the parameters so the flow can run.
-4. Add exceptions for when required fields (`voucherDesc`, `holder`, `recipientParty`) are not found.
-5. Insert a method for finding the notary.
+4. Add appropriate error handling - exceptions must be thrown when required fields (`voucherDesc`, `holder`, `recipientParty`) are not found.
+5. Insert the following method for finding the notary: `notaryLookup.notaryIdentities.<notary name>()`.
```

Comment:
@peterli-r3 We Should not be using `.first ` in the method for getting the notary so I changed `notaryLookup.notaryIdentities.first()` to `notaryLookup.notaryIdentities.<notary name>()` here and in the code. Is that correct or what is the name of the notary I'm looking for?

Reply:
ok. I will change the code.

Reply:
Ummmm. I actually need ask Engineering about it...

Reply:
updated the tutorial with notaryLookup.getNotary(CordaX500Name.parse("O=notary, L=London, C=GB"))!!